Multisig/UI/App/UIKit/TransactionlListViewController/TransactionListViewController.swift

```diff
@@ -0,0 +1,206 @@
+        }
+    }
+
+    private func startNextPageLoadingAnimation() {
+        let indicator = UIActivityIndicatorView(style: .medium)
+        indicator.startAnimating()
+        indicator.frame = CGRect(origin: .zero, size: CGSize(width: tableView.bounds.width, height: 100))
+        // moves the indicator up
+        indicator.bounds = CGRect(origin: CGPoint(x: 0, y: 30), size: indicator.frame.size)
+        tableView.tableFooterView = indicator
+    }
+
+    private func stopNextPageLoadingAnimation() {
+        tableView.tableFooterView = UIView(frame: CGRect(x: 0, y: 0, width: 0, height: 100))
+    }
+
+    private func loadNextPage() {
+        // re-entrancy: if loading already, do not cancel and restart
+        guard let nextPageUri = model.next, loadNextPageDataTask == nil else { return }
+
+        startNextPageLoadingAnimation()
+        do {
+            loadNextPageDataTask = try clientGatewayService.asyncTransactionList(pageUri: nextPageUri) { [weak self] result in
+
+                guard let `self` = self else { return }
+                switch result {
+                case .failure(let error):
+                    DispatchQueue.main.async { [weak self] in
+                        guard let `self` = self else { return }
+                        // ignore cancellation error due to cancelling the
+                        // currently running task. Otherwise user will see
+                        // meaningless message.
+                        if (error as NSError).code == URLError.cancelled.rawValue &&
+                            (error as NSError).domain == NSURLErrorDomain {
+                            return
+                        }
+                        self.onError(error)
+                    }
+                case .success(let page):
+                    var model = TransactionsListViewModel(page.results.flatMap { TransactionViewModel.create(from: $0) })
+                    model.next = page.next
+
+                    DispatchQueue.main.async { [weak self] in
+                        guard let `self` = self else { return }
+                        self.model.append(from: model)
+                        self.onSuccess()
+                    }
+                }
+
+                DispatchQueue.main.async { [weak self] in
+                    self?.stopNextPageLoadingAnimation()
+                }
+
+                self.loadNextPageDataTask = nil
+            }
+
+        } catch {
+            onError(error)
+        }
+    }
+
+    func numberOfSections(in tableView: UITableView) -> Int {
+        model.sections.count
+    }
+
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        model.sections[section].transactions.count
+    }
+
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = tableView.dequeueCell(TransactionListTableViewCell.self, for: indexPath)
+
+        // remove existing controller
```

Comment:
will it be better to put this code to prepareForReuse on cell?

Reply:
I'll move it to a setTransaction method